src/main/java/de/cas/mse/exercise/csv/CsvReader.java

```diff
@@ -1,17 +1,57 @@
 package de.cas.mse.exercise.csv;
 
+import java.io.BufferedReader;
 import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
+import de.cas.mse.exercise.csv.impl.CsvFile;
 import de.cas.mse.exercise.csv.ui.CsvUi;
+import de.cas.mse.exercise.csv.ui.Filterable;
 
-public class CsvReader {
-	
-	private CsvUi csvUi;
+public class CsvReader implements Filterable {
 
-	public void run(final File csvFile) throws Exception {
-		// TODO: Start coding here
-	}
-	
+    private CsvUi csvUi;
+    private CsvFile csvFile;
+
+    public List<List<String>> filter(int col, String word) {
+        return csvFile.contents.stream().filter(row -> row.get(col).contains(word)).collect(Collectors.toList());
```

Comment:
This line is excessively long, also it'd be nicer if `CsvFile` implements the filtering itself

Reply:
We have almost the exact same impl :D

Reply:
My point stand! :D